### PR TITLE
Section on configuring subdataset clone candidates

### LINKED ISSUE #523: Document how subdatasets can be cloned from different URLS
In standard Git there is only a single URL (recorded in `.gitmodules`). However, DataLad has more flexible way to obtain subdatasets.

We need to document how these work, and there is no good place put this info in the sources.

Documentation will come with https://github.com/datalad/datalad/pull/4619

### PR DESCRIPTION
**Pull request by adswa**
fixes #523

### CODE REVIEW

File: docs/beyond_basics/101-148-clonepriority.rst [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+
+Consider a hierarchy of datasets that exist in several locations, for example one :term:`Remote Indexed Archive (RIA) store` *with* a storage special remote [#f2]_, and one without a special remote.
+The topmost superdataset is published to a human-readable and accessible location such as :term:`GitHub` or :term:`GitLab`, and should be configured to always clone subdatasets from the RIA store *with* the storage special remote, even if it was originally created with subdatasets from the RIA store with no storage sibling.
+In order to be able to retrieve subdataset *data* from the subdatasets after cloning the hierarchy of datasets, the RIA store with the storage special remote needs to be configured as a clone candidate.
+Importantly, it should not only be configured as one alternative, but it should be configured as the first location to try to clone from -- else, cloning from the wrong RIA store could succeed and prevent any configured second clone candidate location from being tried.
+
+.. note::
+
+   The most likely use case for such a scenario is in the case of centrally managed data with data administrators that provide and manage the data for its users.
```
Comment (mih) [inline docs/beyond_basics/101-148-clonepriority.rst:57]: s/its/their

File: docs/beyond_basics/101-148-clonepriority.rst [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+    - ``700`` for any unprioritized ``datalad.get.subdataset-source-candidate`` config
+    - ``900`` for the local subdataset path
+
+
+With the ``datalad.get.subdataset-source-candidate`` configuration any number of (differently named) clone candidates can be set and prioritized.
+This allows precise access control over subdataset clone locations, and can -- depending on how many subdataset locations are known and functional -- speed up dataset installation.
+
+.. note::
+
+   Do note how the clone candidate configuration did not specify a concrete dataset in the RIA store, but used the ``{id}`` placeholder, which will expand to the subdataset's :term:`dataset ID` upon cloning.
```
Comment (mih) [inline docs/beyond_basics/101-148-clonepriority.rst:85]: Maybe add some additional info on what placeholders are available. I am pasting the relevant description from the docstring below:

```
    A template string assigned to such a variable can utilize the Python format
    mini language and may reference a number of properties that are inferred
    from the parent dataset's knowledge about the target subdataset. Properties
    include any submodule property specified in the respective `.gitmodules`
    record. For convenience, an existing `datalad-id` record is made available
    under the shortened name `id`.
```

We are likely to extend the list of properties in the future.

**Review by mih — APPROVED** [review]
LGTM! Thanks and sorry for taking so long to review. There is only one fix todo, and possibly a small extension to cover all info comprehensively.

**Comment by adswa** [comment]
Done. Will merge once the tests have passed.

**Pull request merged**